### PR TITLE
book: Move from `pub` to `pub(crate)`

### PR DESCRIPTION
**Pull request by Hofer-Julian**
It is arguably the correct way for this

### CODE REVIEW

File: book/listings/actions/5/window/imp.rs [hunk 1]
```diff
@@ -5,9 +5,9 @@ use gtk::{glib, CompositeTemplate, Label};
 // Object holding the state
 #[derive(CompositeTemplate, Default)]
 #[template(resource = "/org/gtk_rs/example/window.ui")]
-pub struct Window {
```
Comment (bilelmoussaoui) [inline book/listings/actions/5/window/imp.rs:8]: the imp struct could be pub(super) though

File: book/listings/actions/5/window/imp.rs [hunk 2]
```diff
@@ -5,9 +5,9 @@ use gtk::{glib, CompositeTemplate, Label};
...
+pub(crate) struct Window {
     #[template_child]
-    pub label: TemplateChild<Label>,
+    pub(crate) label: TemplateChild<Label>,
```
Comment (bilelmoussaoui) [inline book/listings/actions/5/window/imp.rs:10]: same, it should be pub(crate)

**Comment by Hofer-Julian** [comment]
> Using pub(crate) for a binary doesn't really make sense.

I thought so, too.
Turns out that also for binaries, cargo is not able to detect dead code if it's fully public.

File: book/listings/actions/5/window/imp.rs (hunk 2 above)
Comment (bilelmoussaoui) [reply book/listings/actions/5/window/imp.rs:10]: yes, edited

**Comment by Hofer-Julian** [comment]
> > > Using pub(crate) for a binary doesn't really make sense.
> > 
> > 
> > I thought so, too. Turns out that also for binaries, cargo is not able to detect dead code if it's fully public.
> 
> you can't really detect dead code for gtk apps, especially if you use composite template callbacks

Yeah, just noticed that too. Not sure if it's a macro thing, but static analysis totally fails there.
This makes this PR mostly useless.
Thanks for the comments @bilelmoussaoui!

**Comment by bilelmoussaoui** [comment]
> Yeah, just noticed that too. Not sure if it's a macro thing, but static analysis totally fails there.
This makes this PR mostly useless.

The macro actually uses the functions to create closures from them, but they can only be used at runtime. so nothing can be done at build time...

**Pull request closed without merging**